src/cascadia/TerminalCore/TerminalApi.cpp

```diff
@@ -365,11 +365,23 @@ bool Terminal::EraseInDisplay(const DispatchTypes::EraseType eraseType)
 
 bool Terminal::SetWindowTitle(std::wstring_view title)
 {
+    // Set the title on Terminal load
+    if (_title.empty())
+    {
+        _title = title;
+        _pfnTitleChanged(title);
+    }
+
     _title = title;
 
+    if (_suppressApplicationTitle)

```

Comment:
Couldn't we just _not_ call `_pfnTitleChanged` when `_suppressApplicationTitle` is true? That would negate the need to the `Terminal` core to know about startingTitle, it would just _not_ send title change events at all.

Maybe the `_pfnTitleChanged` needs to be called on the _first_ time through this method, but not on subsequent calls...

Reply:
If we don't call `_pfnTitleChanged` when `_suppressApplicationTitle` is true, refocusing on the tab/pane within the tab (`term.GotFocus()` in TerminalPage.cpp) will call that tab's stored title, which is stored as the application title.

Reply:
It seems like we could just set `_title` to `StartingTitle` as early as possible (when we apply the settings), and then we never have to make the check here on 368-373. Then, we can just simplify the code in here to:

```
if (!_suppressApplicationTitle)
{
    _title = title;
    if (_pfnTitleChanged)
    {
        _pfnTitleChanged(_title)
    }
}
return true;
```

and that's it. We never update `_title`, we never check if it's empty (settings always sets it), we don't have to store `_startingTitle` separately :smile: